app/helpers/search_bar_helper.rb

```diff
@@ -1,32 +1,21 @@
 module SearchBarHelper
-  def mount_search_bar(
-    id,
-    controller: auto_complete_controller_name,
-    url: send("auto_complete_search_#{auto_complete_controller_name}_path"),
-    search_query: params[:search],
-    use_bookmarks: true,
-    use_key_shortcuts: true,
-    autocomplete_id: "searchBar"
-  )
-    bookmarks = {}
-    if use_bookmarks
-      bookmarks = {
-        url: main_app.api_bookmarks_path,
-        canCreate: authorizer.can?(:create_bookmarks),
-        documentationUrl: documentation_url("4.1.5Searching"),
-      }
-    end
-    react_component("SearchBar", {
-                      data: {
-                        controller: controller,
+  def mount_search_bar
+    bookmarks = {
+      url: main_app.api_bookmarks_path,
+      canCreate: authorizer.can?(:create_bookmarks),
+      documentationUrl: documentation_url("4.1.5Searching"),
+    }
+    url = send("auto_complete_search_#{auto_complete_controller_name}_path")
+    react_component("SearchBar", data: {
+                      controller: auto_complete_controller_name,
                       autocomplete: {
-                        id: autocomplete_id,
-                        searchQuery: search_query,
+                        id: 'searchBar',
```

Comment:
I was hoping to get rid of `id` and `useKeyShortcuts` here and rely on the defaults in the component, but looks like react only does shallow merge of default props (meaning that if `autocomplete` hash is present it overrides the contents of the defaults)

Reply:
This is the only thing I could find: https://github.com/facebook/react/issues/2568
But I can try to have a look into it after we merge this, I'd like to get rid of the `data` key as well, so I might send a follow-up.